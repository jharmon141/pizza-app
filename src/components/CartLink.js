import React from 'react'
import { Link } from 'react-router-dom'
import { Col, Row } from 'react-bootstrap'
import PropTypes from 'prop-types'

class CartLink extends React.Component {

   constructor(props) {
      super(props)

      this.state = {
         quantity: 0 
      }
   }

   static propTypes = {
      store: PropTypes.shape({
         pizzas: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.number.isRequired,
            size: PropTypes.string.isRequired,
            basePrice: PropTypes.number.isRequired,
            quantity: PropTypes.number.isRequired,
            toppings: PropTypes.arrayOf(PropTypes.shape({
               __typename: PropTypes.string.isRequired,
               name: PropTypes.string.isRequired,
               price: PropTypes.number.isRequired
            }).isRequired).isRequired
         }).isRequired).isRequired
      })
   }

   componentWillReceiveProps(nextProps){
      if (nextProps.store.pizzas.length > 0) {
         let sum = 0
         nextProps.store.pizzas.map((pizza) => sum += pizza.quantity)
         this.setState({
            quantity: sum
         })
      }
   }

   render() {
      return (
         <Row>
            <Col xs={2} xsOffset={10}>
               <div className="cartLink">
                  <Link to='/cart'>({this.state.quantity})Cart</Link>
               </div>
            </Col>
         </Row>
      )
   }
}

export default CartLink
