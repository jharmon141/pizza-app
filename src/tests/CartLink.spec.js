import React from 'react';
import CartLink from '../components/CartLink';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('CartLink Component', () => {
   it('renders without crashing', () => {
      let quantity = 4;

      const tree = renderer.create(
         <MemoryRouter> 
            <CartLink 
               quantity={4}
            />
         </MemoryRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });
});
