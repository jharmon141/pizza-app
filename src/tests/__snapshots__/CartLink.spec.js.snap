// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartLink Component renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-2 col-xs-offset-10"
  >
    <div
      className="cartLink"
    >
      <a
        href="/cart"
        onClick={[Function]}
      >
        (
        0
        )Cart
      </a>
    </div>
  </div>
</div>
`;
